src/lib/snark_worker/functor.ml

```diff
@@ -104,15 +103,19 @@ module Make (Inputs : Intf.Inputs_intf) :
       with
       | Error e ->
           log_and_retry "getting work" e
-      | Ok None ->
+      | Ok (None, _) ->
```

Comment:
Can you change this to `(Work.Spec.t, Public_key.Compressed.t) option` and don't send work when the key is none?

See issue below